### PR TITLE
Issue#1505 distribution deal service

### LINKED ISSUE #1435: Move +state folder from distribution deal/marketplace to lib/movie/distributiondeal  folder
Since distribution deal is a subcollection of movie, the state folder should be in movie lib, next to the movie folder:
`libs/movie/src/lib/distribution-deal

### PR DESCRIPTION
**Pull request by ioaNikas**
- [x] Move DistributionDeals into the right place and make it a subcollection service. (close #1435 )
- [x] Create a Lib for Contract.
- [ ] Move Cart into and organization subcollection folder.

### CODE REVIEW

File: libs/contract/src/lib/+state/contract.service.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  constructor(store: ContractStore) {
+    super(store);
+  }
+
+  /**
+   * @param contract
+   */
+  public async addContract(contract: Contract): Promise<string> {
+    if (!contract.id) {
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/+state/contract.service.ts]: This is done by akita-ng-fire

File: libs/movie/src/lib/distribution-deals/+state/distribution-deal.query.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Injectable } from '@angular/core';
+import { QueryEntity } from '@datorama/akita';
+import { DistributionDeal } from './distribution-deal.model';
+import { DistributionDealStore, DistributionDealState } from './distribution-deal.store';
+
+@Injectable({
+  providedIn: 'root'
+})
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/distribution-deals/+state/distribution-deal.query.ts]: put injetable on one line : `@Injectable({ providedIn: 'root' })`

File: libs/movie/src/lib/distribution-deals/+state/distribution-deal.service.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import { MovieQuery } from '../../movie/+state/movie.query';
+import { CollectionConfig, CollectionService } from 'akita-ng-fire';
+import { AngularFirestoreCollection } from '@angular/fire/firestore';
+import { DistributionDeal } from './distribution-deal.model';
+import { createContractTitleDetail } from '@blockframes/contract/+state/contract.model';
+import objectHash from 'object-hash';
+import { firestore } from 'firebase';
+
+
+@Injectable({
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/distribution-deals/+state/distribution-deal.service.ts]: same

File: libs/movie/src/lib/distribution-deals/+state/distribution-deal.service.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+  get path() {
+    return `movies/${this.movieQuery.getActiveId()}/distributionDeals`;
+  }
+
+  /**
+   *
+   * @param movieId
+   */
+  private distributionDealsCollection(movieId: string): AngularFirestoreCollection<DistributionDeal> {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/distribution-deals/+state/distribution-deal.service.ts]: just call it `getCollection(movieId: string)`

File: libs/movie/src/lib/distribution-deals/create/create.form.ts [hunk 5]
```diff
@@ -5,12 +5,35 @@ import {
...
+    subtitle: false,
+    ...movieLanguage
+  } as MovieLanguageSpecification;
+}
+
+export function createLanguageControl(
+  language: MovieLanguageSpecification,
+  disableDubbed?: boolean
+) {
+  return new FormGroup({
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/distribution-deals/create/create.form.ts]: Hmm, I don't think this is how it work. A `create***Control` should return the content of the form group but not the FormGroup itself. Else you cannot use it for typings
```typescript
return {
    original: new FormControl(language.original),
    dubbed: new FormControl({ value: language.dubbed, disabled: disableDubbed }),
    subtitle: new FormControl(language.subtitle)
}
```

File: tsconfig.json [hunk 6]
```diff
@@ -49,40 +54,58 @@
...
+        "libs/movie/src/lib/movie/*"
       ],
       "@blockframes/movie/import": [
-        "libs/movie/src/lib/components/import/import-movie.module.ts"
+        "libs/movie/src/lib/movie/components/import/import-movie.module.ts"
+      ],
+      "@blockframes/movie/types": [
+        "libs/movie/src/lib/movie/+state/movie.firestore.ts"
       ],
+      "@blockframes/movie/deals/*": [
```
Comment (GrandSchtroumpf) [inline tsconfig.json]: You should keep the same principles than movie. Either you call it `@blockframes/distribution-deal` or you call movie : `@blockframes/movie/movie`.

File: libs/contract/src/lib/+state/contract.service.ts (hunk 1 above)
Comment (phpgeekfr) [reply libs/contract/src/lib/+state/contract.service.ts]: can we focus on moving libs only on this PR ? Because this will lead to hard rebase.

Comment (GrandSchtroumpf) [reply libs/contract/src/lib/+state/contract.service.ts]: Yes, good for me. Maybe we should have done this in three PR then, as I thought it was part of `Create Lib for Contract`

**Pull request merged**